include/exadg/incompressible_navier_stokes/spatial_discretization/operator_consistent_splitting.cpp

```diff
@@ -0,0 +1,776 @@
+              velocity_minus.get_gradient(q) * velocity_minus.get_value(q);
+            const auto gradu_u_plus =
+              velocity_plus.get_gradient(q) * velocity_plus.get_value(q);
+            const auto convective_flux =
+              Number(0.5) * (gradu_u_minus + gradu_u_plus) * normal;
+
+            pressure_minus.submit_value(convective_flux, q);
+            pressure_plus.submit_value(-convective_flux, q);
+          }
+
+        pressure_minus.integrate_scatter(dealii::EvaluationFlags::values, dst);
+        pressure_plus.integrate_scatter(dealii::EvaluationFlags::values, dst);
+      }
+}
+
+
+template<int dim, typename Number>
+void
+OperatorConsistentSplitting<dim, Number>::local_rhs_ppe_div_term_convective_boundary_face(
+  dealii::MatrixFree<dim, Number> const & matrix_free,
+  VectorType &                            dst,
+  VectorType const &src,
+  Range const & face_range) const
+{
+  unsigned int dof_index_velocity  = this->get_dof_index_velocity();
+  unsigned int dof_index_pressure  = this->get_dof_index_pressure();
+  unsigned int quad_index_overintegration = this->get_quad_index_velocity_overintegration();
+
+  FaceIntegratorP pressure_minus(matrix_free, true, dof_index_pressure, quad_index_overintegration);
+  FaceIntegratorU velocity_minus(matrix_free, true, dof_index_velocity, quad_index_overintegration);
+
+  for (unsigned int face = face_range.first; face < face_range.second; face++)
+    {
+      dealii::types::boundary_id const boundary_id = matrix_free.get_boundary_id(face);
+
+      BoundaryTypeU const boundary_type =
+        this->boundary_descriptor->velocity->get_boundary_type(boundary_id);
+
+ 
+
+  if(boundary_type == BoundaryTypeU::Dirichlet or boundary_type == BoundaryTypeU::DirichletCached)
+  {
+          pressure_minus.reinit(face);
+          for (const unsigned int q : pressure_minus.quadrature_point_indices())
+            {
+              // Cancels with consistent boundary condition
+              pressure_minus.submit_value({}, q);                
+            }
+
+          pressure_minus.integrate_scatter(dealii::EvaluationFlags::values, dst);
+    }
+    else
+        {
+          pressure_minus.reinit(face);
+          velocity_minus.reinit(face);
+
+          velocity_minus.gather_evaluate(src, dealii::EvaluationFlags::values | dealii::EvaluationFlags::gradients);
+
+          for (const unsigned int q : pressure_minus.quadrature_point_indices())
+            {
+              const auto normal = pressure_minus.normal_vector(q);
+
+              // auto bc = this->boundary_descriptor->velocity->neumann_bc.find(boundary_id)->second;
+              // auto q_points = pressure_minus.quadrature_point(q);
+              // This should be the boundary condition but we get only grad_u * n which we cannot use, so use the gradient instead
+              const auto  grad_u = velocity_minus.get_gradient(q); //FunctionEvaluator<1, dim, Number>::value(*bc, q_points, this->evaluation_time);
```

Comment:
this comment is a bit unclear, can you rephrase?
The point is that you cannot insert the boundary data, as we have only $\nabla p \cdot \mathbf{n}$, right?
Maybe consider something like
"The provided boundary data cannot be used to compute the flux, use the elements velocity instead."